### PR TITLE
fix: open new tab on companion for login and auth v3

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Fixes the opening of a new tab for when the companion is active
- Fixes the issue for v3 to always open new tab

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-618 #done
WT-620 #done

### CODE REVIEW

File: packages/shared/src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -113,7 +122,15 @@ export const AuthContextProvider = ({
       isFirstVisit: user?.isFirstVisit ?? false,
       trackingId: user?.id,
       shouldShowLogin: loginState !== null,
-      showLogin: (trigger) => setLoginState({ trigger }),
+      showLogin: (trigger) => {
+        const hasCompanion = !!document.querySelector('daily-companion-app');
+        if (hasCompanion || authVersion === AuthVersion.V3) {
+          const signup = `${process.env.NEXT_PUBLIC_WEBAPP_URL}/signup?close=true`;
```
Comment (rebelchris) [inline packages/shared/src/contexts/AuthContext.tsx:128]: When you use the close one it auto-closes the page, in that case you should opt to open in new tab.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice one, just one small remark

File: packages/shared/src/contexts/AuthContext.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/contexts/AuthContext.tsx:128]: We are using `window.open` already. That opens a new tab.
https://github.com/dailydotdev/apps/blob/915705ae6c71b007c5c89d8a08a519c59ef8f0d1/packages/shared/src/contexts/AuthContext.tsx#L129

Comment (rebelchris) [reply packages/shared/src/contexts/AuthContext.tsx]: Ah right, forgot `_blank` is the default 🙏

**Review by rebelchris — APPROVED** [review]
Ready 🚀

**Pull request merged**